filters/apimonitoring/filter.go

```diff
@@ -0,0 +1,158 @@
+package apimonitoring
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const (
+	metricsSeparator = "."
+	metricCountAll   = "http_count"
+	metricCount500s  = "http5xx_count"
+	metricCount400s  = "http4xx_count"
+	metricCount300s  = "http3xx_count"
+	metricCount200s  = "http2xx_count"
+	metricLatency    = "latency"
+)
+
+const (
+	stateBagKeyPrefix = "filter.apimonitoring."
+	stateBagKeyState  = stateBagKeyPrefix + "state"
+)
+
+var (
+	unknownPath = &pathInfo{
+		ApplicationId: unknownElementPlaceholder,
+		ApiId:         unknownElementPlaceholder,
+		PathTemplate:  unknownElementPlaceholder,
+	}
+)
+
+type apiMonitoringFilter struct {
+	paths []*pathInfo
+}
+
+type pathInfo struct {
+	ApplicationId string
+	ApiId         string
+	PathTemplate  string
+	Matcher       *regexp.Regexp
+}
+
+// apiMonitoringFilterContext contains information about the metrics tracking
+// for one HTTP exchange (one routing). It serves to pass information from
+// the `Request` call to the `Response` call (stored in the context's `StateBag`).
+type apiMonitoringFilterContext struct {
+	// DimensionPrefix is the prefix to all metrics tracked during this exchange (generated only once)
+	DimensionsPrefix string
+	// Begin is the earliest point in time where the request is observed
+	Begin time.Time
+	// OriginalRequestSize is the initial requests' size, before it is modified by other filters.
+	OriginalRequestSize int64
+}
+
+func (f *apiMonitoringFilter) String() string {
+	return toJsonStringOrError(mapApiMonitoringFilter(f))
+}
+
+func (f *apiMonitoringFilter) Request(c filters.FilterContext) {
+	log := log.WithField("op", "request")
```

Comment:
I would drop this. This is way to much log information and majorly unimportant although for debugging.

Reply:
That only adds a field on the next log entries. That does not log anything in itself.
Just want to be sure we understand the same thing.

It can indeed be removed if you judge it too much.

Reply:
Solved in 8a62d0d80d0e4e154a9801a73e3234c99f1bcb0e